instance/models/instance.py

```diff
@@ -134,6 +152,68 @@ def status(self):
         else:
             return active_server_set[0].status
 
+    @property
+    def event_context(self):
+        """
+        Context dictionary to include in events
+        """
+        return {'instance_id': self.pk}
+
+    @staticmethod
+    def on_post_save(sender, instance, created, **kwargs):
+        """
+        Triggered by the post_save event, with `created` a boolean indicating if the object was just
+        created in the DB
+        """
+        self = instance
+        self.update_logger()
+
+    def update_logger(self):
+        """
+        Start referencing the instance in logs once the DB entry exist
+        """
+        if self.pk is not None:
+            self.logger = InstanceLoggerAdapter(logger, {'obj': self})
+
+    @property
+    def log_text(self):
+        """
+        Combines the instance and server log outputs in chronological order
+        Currently only supports one non-terminated server at a time
+        Returned as a text string
+        """
+        current_server = self.active_server_set.get()
+        # TODO: Filter out log entries for which the user doesn't have view rights
+        server_log_entry_set = current_server.log_entry_set.order_by('pk')\
+                                                           .iterator()
+        instance_log_entry_set = self.log_entry_set.order_by('pk')\
+                                                   .iterator()
+
+        next_server_log_entry = partial(next, server_log_entry_set, None)
+        next_instance_log_entry = partial(next, instance_log_entry_set, None)
+
+        log_text = ''
+        instance_log_entry = next_instance_log_entry()
+        server_log_entry = next_server_log_entry()
+
+        while instance_log_entry is not None and server_log_entry is not None:
+            if server_log_entry.created < instance_log_entry.created:
+                log_text += '{}\n'.format(server_log_entry)
+                server_log_entry = next_server_log_entry()
+            else:
+                log_text += '{}\n'.format(instance_log_entry)
+                instance_log_entry = next_instance_log_entry()
+
+        while instance_log_entry is not None:
+            log_text += '{}\n'.format(instance_log_entry)
+            instance_log_entry = next_instance_log_entry()
+
+        while server_log_entry is not None:
+            log_text += '{}\n'.format(server_log_entry)
+            server_log_entry = next_server_log_entry()
+
+        return log_text
```

Comment:
@mtyaka I'm not sure about this change. On on side it looks better, having a unique table makes things simpler - it avoids the type of logic we're seeing here, and would allow to group 3 different types of objects into one. But on the other side, it makes some parts harder to evolve - for example, if we add a new (different) type of server, like say an `AWSServer`, we have to create a brand new table, or add a superfluous column. Same thing with adding different types of instances. Or simply if we want to add DB logging for another object - it will be harder to share the internal plumbing.

You might end up being right, but I'd like to see how we are going to deal with things like https://opencraft.uservoice.com/forums/280754-opencraft/suggestions/9937191-self-deployment or https://opencraft.uservoice.com/forums/280754-opencraft/suggestions/9942105-add-support-for-aws-servers before deciding to change.